pandas/core/tools/datetimes.py

```diff
@@ -334,6 +334,9 @@ def _convert_listlike_datetimes(
                 return DatetimeIndex(arg, tz=tz, name=name)
             except ValueError:
                 pass
+        elif tz:
```

Comment:
I think this would fail if a user passes `to_datetime(np.array([...], dtype='datetime64[ns]'), utc=True)`?

I was thinking more:

```
if box:
    if not isinstance(arg, (DatetimeArray, ...)):
        ....
    elif tz:
        ...
```

Reply:
```
>>> arr = np.arange(2).astype('datetime64[ns]')
>>> to_datetime(arr, utc=True)
DatetimeIndex(['1970-01-01 00:00:00+00:00', '1970-01-01 00:00:00.000000001+00:00'], dtype='datetime64[ns, UTC]', freq=None)
```

or are you suggesting that this is wrong because `box=True` was not passed?

Reply:
Oh! It does work. `box=True` is the default so this looks correct.